pkg/datapath/iptables/reconciler.go

```diff
@@ -145,6 +145,17 @@ func reconciliationLoop(
 
 	firstInit := true
 
+	// Run an initial full reconciliation before listening on partial reconciliation
+	// request channels (like proxies and no track rules).
+	if err := updateRules(state, firstInit); err != nil {
+		health.Degraded("iptables rules update failed", err)
```

Comment:
Is there direct correlation between health and iptable rule updates failure ? iptable records might be too much, the update could be just delayed. 
what you think about adding "updateErrors" count in one structure and increase the count for this case ?

Reply:
> Is there direct correlation between health and iptable rule updates failure ? iptable records might be too much, the update could be just delayed.

After triggering a reconciliation, the reconciler waits until the rules are either installed successfully or an error is reported. Even if it takes a lot of time, the reconciler waits to report the correct status through the health reporter.
The clients requesting a reconciliation have a way to wait for it synchronously through the notification in the `reconciliationRequest` channel, so I'd say that longer reconciliation time are not an issue.

> what you think about adding "updateErrors" count in one structure and increase the count for this case ?

I think we already have something like this, take a look at the `cilium-dbg statedb health` command output.